components/support/base/src/main/java/mozilla/components/support/base/ids/NotificationIds.kt

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.base.ids
+
+import android.app.Notification
+import android.app.NotificationManager
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.core.app.NotificationManagerCompat
+
+// If the tag is not used again in one week then clear the id
+private const val NOTIFICATION_ID_LIFETIME: Long = 1000 * 60 * 60 * 24 * 7
+
+// We start at 10000 instead of 0 to avoid conflicts with app code using random low numbers.
+private const val ID_OFFSET = 10000
+
+private const val PREFERENCE_FILE = "notification_ids"
+
+/**
+ * Helper for component and app code to use unique notification ids without conflicts.
+ */
+object NotificationIds {
+    private val ids = SharedIds(PREFERENCE_FILE, NOTIFICATION_ID_LIFETIME, ID_OFFSET)
+
+    /**
+     * Get a unique notification ID for the provided unique tag.
+     */
+    fun getIdForTag(context: Context, tag: String): Int = ids.getIdForTag(context, tag)
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal var now: () -> Long
+        get() = ids.now
+        set(value) { ids.now = value }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun clear(context: Context) = ids.clear(context)
+}
+
+/**
+ * Post a notification to be shown in the status bar.
+ *
+ * Uses a unique [String] tag instead of an [Int] id like [NotificationManager.notify].
+ */
+fun NotificationManager.notify(context: Context, tag: String, notification: Notification) {
```

Comment:
Could be helpful to also mention the life span of the unique id?

Reply:
I consider that an implementation detail that may change.

Reply:
Then should we also return the id? Because as consumer ,I could assumed that the unique id is not going to change for this tag.

Reply:
If you want the ID then you can use `NotificationIds` yourself. Those are extension methods since you do not need to care about them when you just want to `notify()` and `cancel()`. The ids doesn't change for a tag (unless expired..).

Reply:
What I meant, was that we are not communicating that the ids can get expired  neither here nor on `NotificationIds.getIdForTag`

Reply:
I don't think the consumer necessarily needs to care. This is mostly to not have an indefinitely growing file in the app directory. If a component needs to show a notification again one month later then it's fine if it gets a new id assigned. It's only critical when updating/cancelling/overriding/conflicting with an existing notification?

Reply:
👍